### PR TITLE
Improvements to documentation deploy process

### PR DESCRIPTION
**Pull request by samkearney**
- Build documentation from pull requests and publish them to a staging directory
- Capture warnings from documentation build in AzUre Pipelines

### CODE REVIEW

File: src/rdmnet/broker/broker_api.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
```
Comment (nwagner) [inline src/rdmnet/broker/broker_api.cpp]: It's getting to be the end of the year.  We'll want to remember to update the copyright to 2020 when we're touching files.

File: src/rdmnet/broker/broker_threads.h [hunk 2]
```diff
@@ -17,9 +17,8 @@
  * https://github.com/ETCLabs/RDMnet
  *****************************************************************************/
 
-/// \file broker_threads.h
-/// \brief Classes to represent threads used by the Broker.
-/// \author Nick Ballhorn-Wagner and Sam Kearney
+// broker_threads.h
```
Comment (nwagner) [inline src/rdmnet/broker/broker_threads.h]: Sure, all credit taken away.  :)

File: src/rdmnet/broker/broker_api.cpp (hunk 1 above)
Comment (samkearney) [reply src/rdmnet/broker/broker_api.cpp]: I was just envisioning scripting a mass update rather than touching files individually.

Comment (nwagner) [reply src/rdmnet/broker/broker_api.cpp]: Agreed.  This was a convenient spot to make sure you were reminded. :)

**Pull request merged**